python/pyspark/sql/readwriter.py

```diff
@@ -373,6 +373,12 @@ def csv(self, path, schema=None, sep=None, encoding=None, quote=None, escape=Non
                        default value, ``false``.
         :param inferSchema: infers the input schema automatically from data. It requires one extra
                        pass over the data. If None is set, it uses the default value, ``false``.
+        :param enforceSchema: If it is set to ``true``, the specified or inferred schema will be
+                              forcibly applied to datasource files and headers in CSV files will be
+                              ignored. If the option is set to ``false``, the schema will be
```

Comment:
Is it _ignored_?

Reply:
Same confusion from https://github.com/apache/spark/pull/20894/files#r188553979.

Reply:
@rxin @gatorsmile `enforceSchema` confuses people. Should I ignore `spark-api` decision and rename it back to `checkHeader`?

Reply:
We should not silently ignore the option. When `header`  is `false` and `enforceSchema`, we should issue an exception.